app/src/main/java/com/onewelcome/showcaseapp/ui/screens/HomeScreen.kt

```diff
@@ -1,36 +1,80 @@
 package com.onewelcome.showcaseapp.ui.screens
 
-import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.Image
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Card
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.Alignment
+import androidx.compose.runtime.ReadOnlyComposable
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
 import androidx.navigation.NavController
-import com.onewelcome.showcaseapp.ui.theme.ShowcaseAppTheme
+import com.onewelcome.showcaseapp.R
+import com.onewelcome.showcaseapp.navigation.ScreenNavigation
+import com.onewelcome.showcaseapp.ui.screens.sections.SectionItem
 
 @Composable
 fun HomeScreen(navController: NavController) {
+  HomeScreenContent(onNavigateToSection = { navController.navigate(it) })
+}
```

Comment:
I modified it to be able to have content independent of viewModel. This way we can use @Preview without any issues. 
In case there are more callbacks we can create some callback class that will hold all of the callback for particular screen or we can have an interface to would be implemented by view model.
Let me know what you think about this

Reply:
It's looking nice 👍